tools/cli/commands/build.js

```diff
@@ -793,10 +827,15 @@ async function buildProject( t ) {
 		for ( const key of depTypes ) {
 			if ( packageJson[ key ] ) {
 				for ( const [ pkg, ver ] of Object.entries( packageJson[ key ] ) ) {
-					if ( ver.startsWith( 'workspace:* || ' ) ) {
-						packageJson[ key ][ pkg ] = ver.substring( 15 );
-					} else if ( ver === 'workspace:*' ) {
-						delete packageJson[ key ][ pkg ];
+					if ( ver === 'workspace:*' ) {
+						for ( const ctxPkg of Object.values( t.ctx.versions ) ) {
+							if ( ctxPkg.name === pkg ) {
```

Comment:
This isn't quite right, `ctxPkg.name` is the name from composer.json but for JS packages we need to compare with the name from package.json instead. You'll probably want to put that in as `jsName` or something below (near line 899) so you can use it here.

Its fine for that `jsName` field to be undefined or null or whatever for things that don't have a package.json, of course.

Reply:
Nice catch, [7470c66](https://github.com/Automattic/jetpack/pull/27043/commits/7470c66236f855304b48eeeede791e0babc4689a)